resource-validator/src/main/scala/com/broadinstitute/dsp/resourceValidator/DataprocWorkerChecker.scala

```diff
@@ -37,47 +38,63 @@ object DataprocWorkerChecker {
                       )
-                      .as[Option[RuntimeWithWorkers]](Some(runtime))
-                else
-                  // Here we log a metric if we detect the aforementioned unfixable anomaly. This metric is in addition to the one already reported for finding a worker mismatch
-                  deps.checkRunnerDeps.metrics.incrementCounter(s"$appName/$unfixableAnomalyCheckType") >>
-                    logger
-                      .warn(
-                        s"${runtime} has an anomaly with the number of workers in google. Unable to fix the anomaly. Recording a metric and moving on"
-                      )
-                      .as[Option[RuntimeWithWorkers]](
-                        Some(runtime)
-                      )
-              case false => F.pure(None)
+                      .as(Option(runtime))
+                  // If the number of primary workers is less than 2, modifying workers requires a creation and deletion. Leo does not handles these, so we will not either
+                  case false if c.getConfig.getWorkerConfig.getNumInstances >= 2 =>
+                    if (runtime.r.status.toLowerCase == "running")
+                      deps.dataprocService
+                        .resizeCluster(
+                          runtime.r.googleProject,
+                          regionName,
+                          DataprocClusterName(runtime.r.runtimeName),
+                          if (doesPrimaryWorkerMatch) None else Some(runtime.workerConfig.numberOfWorkers.getOrElse(0)),
+                          if (doesSecondaryWorkerMatch) None
+                          else Some(runtime.workerConfig.numberOfPreemptibleWorkers.getOrElse(0))
+                        )
+                        .void
+                        .handleErrorWith {
+                          case e: com.google.api.gax.rpc.ApiException =>
+                            logger.warn(
+                              s"${runtime} has an anomaly with the number of workers in google, and the resize failed."
+                            ) >> deps.checkRunnerDeps.metrics.incrementCounter(s"$appName/$checkType/failure")
+                        } >>
+                        logger
+                          .warn(
+                            s"${runtime} has an anomaly with the number of workers in google. \n\tPrimary work match status: $doesPrimaryWorkerMatch\n\tSecondary worker match status: ${doesSecondaryWorkerMatch}"
+                          )
+                          .as(Option(runtime))
+                    else
+                      logger
+                        .warn(
+                          s"${runtime} has an anomaly with the number of workers in google, but we cannot attempt to fix it because the cluster is not running. \n\tprimary in google: ${c.getConfig.getWorkerConfig.getNumInstances} secondary in google ${c.getConfig.getSecondaryWorkerConfig.getNumInstances}."
```

Comment:
Should this be
```suggestion
                          s"${runtime} has an anomaly with the number of workers in google, but we cannot attempt to fix it because the cluster is not running. \n\tNumber of primary workers in google: ${c.getConfig.getWorkerConfig.getNumInstances} Number of secondary workers in google: ${c.getConfig.getSecondaryWorkerConfig.getNumInstances}."
```
?

Reply:
Let's make this change before merging.

Reply:
Oops, i committed this but didn't push, thanks.